pkg/lane/quic_test.go

```diff
@@ -0,0 +1,182 @@
+	headerBuffer := make([]byte, 0)
+	header.Pack(&headerBuffer)
+	err1=errors.New("Error")
+	tests := []struct {
+		name    string
+		stream        quic.Stream
+		msg *model.Message
+		wantErr error
+	}{
+		{
+			name:"Test-FailureCase",
+			stream:mockStream,
+			msg:&model.Message{},
+			wantErr:err1,
+		},
+		{
+			name:"Test-SuccessCase",
+			stream:mockStream,
+			msg:&model.Message{Content:"message"},
+			wantErr:nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &QuicLane{
+				stream:        tt.stream,
+			}
+			if ( tt.name == "Test-FailureCase") {
+				mockStream.EXPECT().Read(gomock.Any()).Return(0, err1).Times(1)
+			}
+			if ( tt.name == "Test-SuccessCase") {
+			callFirst:=	mockStream.EXPECT().Read(gomock.Any()).SetArg(0, headerBuffer).Return(packer.HeaderSize,nil).Times(1)
+			mockStream.EXPECT().Read(gomock.Any()).SetArg(0, bytesMsg).Return(len(bytesMsg), nil).Times(1).After(callFirst)
+			}
+			if err := l.ReadMessage(tt.msg); (err != tt.wantErr) {
+				t.Errorf("QuicLane.ReadMessage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// TestQuicLane_WriteMessage is function to test WriteMessage().
+func TestQuicLane_WriteMessage(t *testing.T) {
+	tests := []struct {
+		name    string
+		stream        quic.Stream
+		msg *model.Message
+		wantErr bool
+	}{
+		{
+			name:"TestWriteMessage",
+			stream:mockStream,
+			msg:&model.Message{Content:"message"},
+			wantErr:false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &QuicLane{
+				stream:        tt.stream,
+			}
+			mockStream.EXPECT().Write(gomock.Any()).Return(1,nil).AnyTimes()
+			if err := l.WriteMessage(tt.msg); (err != nil) != tt.wantErr {
+				t.Errorf("QuicLane.WriteMessage() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// TestQuicLane_SetReadDeadline is function to test SetReadDeadline().
+func TestQuicLane_SetReadDeadline(t *testing.T) {
+	tests := []struct {
+		name    string
+		readDeadline  time.Time
+		stream        quic.Stream
+		t time.Time
+		wantErr bool
+	}{
+		{
+			name:"TestSetReadDeadline",
+			readDeadline:time.Time{},
+			t:time.Time{},
+			wantErr:false,
+			stream:mockStream,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &QuicLane{
+				readDeadline:  tt.readDeadline,
+				stream:        tt.stream,
+			}
+			mockStream.EXPECT().SetReadDeadline(gomock.Any()).Return(nil).Times(1)
+			if err := l.SetReadDeadline(tt.t); (err != nil) != tt.wantErr {
+				t.Errorf("QuicLane.SetReadDeadline() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+// TestQuicLane_SetWriteDeadline is function to test SetWriteDeadline().
+func TestQuicLane_SetWriteDeadline(t *testing.T) {
+	tests := []struct {
+		name    string
+		writeDeadline time.Time
+		stream        quic.Stream
+		t time.Time
+		wantErr bool
+	}{
+		{
+			name:"Test1",
```

Comment:
Please give a name that desribes the test in short.
Applicable to similar places.